python/cudaq/runtime/compile.py

```diff
@@ -0,0 +1,46 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+from ..mlir._mlir_libs._quakeDialects import cudaq_runtime
+from ..mlir import ir as mlir
+from ..mlir.dialects import func as func
+
+from cudaq.kernel.kernel_builder import PyKernel
+
+def compile(kernel : PyKernel , *args):
+    ctx = cudaq_runtime.ExecutionContext("compile")
+    cudaq_runtime.setExecutionContext(ctx)
+
+    kernel(*args)
+
+    cudaq_runtime.resetExecutionContext()
+
+    assert(len(ctx.codes) == 1)
+
+    newKernel = PyKernel([])
+    newModule = mlir.Module.parse(asm = ctx.codes[0], context=newKernel.ctx)
+
+    #TODO: implement some clone mechanism
```

Comment:
Python has already a deepcopy function in the copy module that works for any object. All you have to do is:
form copy import deepcopy
newKernel = deepcopy(kernel)